### PR TITLE
SDK update to v1.38.52 to add LogDelivery support

### PR DESCRIPTION
**Pull request by nmvk**
Updated SDK to v1.38.52 so generated code contains new functionalities. Added log delivery to RG coverage test and fixed an issue related to auth token.

```
[gw0] [100%] PASSED tests/test_replicationgroup.py::TestReplicationGroup::test_rg_input_coverage 
================= 1 passed in 1275.99s (0:21:15) =============================
```

Better experience of LogDelivery would probably need https://github.com/aws-controllers-k8s/community/issues/814.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
We can't run e2e test yet, but i'm fine with merging the PR.
/approve

File: generator.yaml [hunk 1]
```diff
@@ -135,3 +135,6 @@ ignore:
     - ModifyReplicationGroupInput.SecurityGroupIds
     - ModifyReplicationGroupInput.EngineVersion
     - CreateReplicationGroupInput.GlobalReplicationGroupId
```
Comment (kumargauravsharma) [inline generator.yaml:137]: There are more fields for log delivery in the output than what are there in the input (Spec).
Those fields are ([api reference](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/elasticache/create-replication-group.html)):
```
Status -> (string)
    Returns the log delivery configuration status. Values are one of enabling | disabling | modifying | active | error
Message -> (string)
    Returns an error message for the log delivery configuration.
```
Both of these fields provide useful information about actual status of this configuration.
As Spec values of saved K8s resource are not updated from the API output, the information contained in Status, and Message fields will not surface.

Thus, consider:
1) adding custom ko.Status fields to replication group:
```
LogDeliveryConfigurations:
    is_read_only: true
    from:
        operation: DescribeReplicationGroups
        path:LogDeliveryConfigurations
```
2) Consider setting resource sync conditions based on `LogDeliveryConfigurations.Status` value by checking it against value `active`
3) Consider setting condition message from `LogDeliveryConfigurations.Message` when `LogDeliveryConfigurations.Status` is `error`

File: test/e2e/resources/replicationgroup_input_coverage.yaml [hunk 2]
```diff
@@ -4,7 +4,6 @@ metadata:
   name: $RG_ID
 spec:
   atRestEncryptionEnabled: true
-  authToken: testplaintexttoken
```
Comment (kumargauravsharma) [inline test/e2e/resources/replicationgroup_input_coverage.yaml:7]: Does its removal impact any e2e test?

File: test/e2e/service_cleanup.py [hunk 3]
```diff
@@ -60,10 +60,17 @@ def delete_snapshot(snapshot_name: str):
...
 def delete_non_default_user(user_id: str):
     ec = boto3.client("elasticache")
     ec.delete_user(UserId=user_id)
     logging.info(f"Deleted non default user {user_id}")
+=======
+def delete_log_group(log_group_name: str):
+    logs = boto3.client("logs")
+    logs.delete_log_group(logGroupName=log_group_name)
+    logging.info(f"Deleted CW log group {log_group_name}")
+>>>>>>> Add LogDelivery to RG Coverage
```
Comment (kumargauravsharma) [inline test/e2e/service_cleanup.py:74]: I think this needs resolving git conflicts

**Review by kumargauravsharma — CHANGES_REQUESTED** [review]

File: test/e2e/service_cleanup.py (hunk 3 above)
Comment (nmvk) [reply test/e2e/service_cleanup.py]: Ah :( Thanks!

**Review by echen-98 — APPROVED** [review]

**Review by kumargauravsharma — APPROVED** [review]

**Pull request merged**